app/addons/activetasks/components.react.jsx

```diff
@@ -0,0 +1,387 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  "app/helpers",
+  "api",
+  "react",
+  "addons/activetasks/stores",
+  "addons/activetasks/resources",
+  "addons/activetasks/actions"
+
+], function (Helpers, FauxtonAPI, React, Stores, Resources, Actions) {
+  var activeTasksStore = Stores.activeTasksStore;
+
+  var ActiveTasksController = React.createClass({
+
+    getStoreState: function () {
+      return {  
+        selectedTab: activeTasksStore.getSelectedTab(),
+        collection: activeTasksStore.getCollection(),
+        searchTerm: activeTasksStore.getSearchTerm(),
+        setPolling: activeTasksStore.setPolling(),
+        clearPolling: activeTasksStore.clearPolling
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    componentDidMount: function () {
+      this.state.setPolling;
+      activeTasksStore.on('change', this.onChange, this);      
+    },
+
+    componentWillUnmount: function() {
+      this.state.clearPolling();
+      activeTasksStore.off('change', this.onChange, this); 
+    },
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+    render: function () {
+      var collection = this.state.collection; 
+      var searchTerm = this.state.searchTerm;
+      var selectedTab = this.state.selectedTab;
+
+      if (collection.length === 0 ) {
+        return ( <div className="active-tasks"><tr><td><p>  No tasks. </p></td></tr></div> );
+      } else {
+        return (
+          <div className="scrollable">
+            <div className="inner">
+              <ActiveTasksFilterHeader/>
+              <ActiveTaskTable collection={collection} searchTerm={searchTerm} tab={selectedTab}/>
+            </div>
+          </div>
+        );
+      }
+    }
+  });
+
+  var ActiveTasksFilterHeader = React.createClass({ //This is for the little '+ Filter' Tab 
+    getStoreState: function () {
+      return {  
+        searchTerm: activeTasksStore.getSearchTerm()
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    toggleFilter: function () {
+      $('#dashboard-content').scrollTop(0);
+      $('#query').toggle('slow');
+    },
+    setNewSearchTerm: function (e) {
+      this.setState({searchTerm: e.target.value});
```

Comment:
If you are storing the searchTerm in the store then you don't need to store it in the state.